cli/build.js

```diff
@@ -114,45 +118,65 @@ function cleanupAot() {
   fs.removeSync(skyPagesConfigUtil.spaPathTemp());
 }
 
-/**
- * Executes the build command.
- * @name build
- */
-function build(argv, skyPagesConfig, webpack) {
-  const compileModeIsAoT = skyPagesConfig &&
-    skyPagesConfig.skyux &&
-    skyPagesConfig.skyux.compileMode === 'aot';
-
-  let buildConfig;
+function buildServe(argv, skyPagesConfig, webpack, isAot) {
+  server.start(skyPagesConfigUtil.getAppBase(skyPagesConfig))
+    .then(port => {
+      argv.assets = argv.assets || `https://localhost:${port}`;
+      buildPromise(argv, skyPagesConfig, webpack, isAot)
+        .then(stats => browser(argv, skyPagesConfig, stats, port));
+    });
+}
 
+function buildPromise(argv, skyPagesConfig, webpack, isAot) {
   const assetsBaseUrl = argv.assets || '';
   const assetsRel = argv.assetsrel;
 
-  const lintResult = tsLinter.lintSync();
-  if (lintResult.exitCode > 0) {
-    process.exit(lintResult.exitCode);
-    return;
-  }
+  let buildConfig;
 
-  if (compileModeIsAoT) {
+  if (isAot) {
     stageAot(skyPagesConfig, assetsBaseUrl, assetsRel);
     buildConfig = require('../config/webpack/build-aot.webpack.config');
   } else {
     buildConfig = require('../config/webpack/build.webpack.config');
   }
 
-  const config = buildConfig.getWebpackConfig(skyPagesConfig);
+  return new Promise((resolve, reject) => {
+    const config = buildConfig.getWebpackConfig(skyPagesConfig);
+    assetsProcessor.setSkyAssetsLoaderUrl(config, skyPagesConfig, assetsBaseUrl, assetsRel);
 
-  assetsProcessor.setSkyAssetsLoaderUrl(config, skyPagesConfig, assetsBaseUrl, assetsRel);
+    runCompiler(webpack, config, isAot)
+      .then(stats => {
+        if (isAot) {
+          cleanupAot();
+        }
 
-  return runCompiler(webpack, config)
-    .then(stats => {
-      if (compileModeIsAoT) {
-        cleanupAot();
-      }
+        resolve(stats);
+      })
+      .catch(reject);
+  });
+}
 
-      return Promise.resolve(stats);
-    });
+/**
+ * Executes the build command.
+ * @name build
+ * @param {*} skyPagesConfig
+ * @param {*} webpack
+ * @param {*} isAot
+ */
+function build(argv, skyPagesConfig, webpack) {
+
+  const lintResult = tsLinter.lintSync();
+  const isAot = skyPagesConfig &&
+    skyPagesConfig.skyux &&
+    skyPagesConfig.skyux.compileMode === 'aot';
+
+  if (lintResult.exitCode > 0) {
+    process.exit(lintResult.exitCode);
+  } else if (argv.serve) {
+    buildServe(argv, skyPagesConfig, webpack, isAot);
+  } else {
+    return buildPromise(argv, skyPagesConfig, webpack, isAot);
```

Comment:
If `skyux build` fails outside of linting, [the ProcessExitCode plugin](https://github.com/blackbaud/skyux-builder/blob/ecce9dba359157f7e7521b05eaf3135d491d189f/config/webpack/common.webpack.config.js#L145) calls `process.exit`.

I was really only concerned with returning a promise from `buildPromise` knowing that `cli/e2e.js` imports `build`, which wouldn't be passing the `serve` command.  It definitely makes sense to make the signature the same (always returning a promise) regardless of the arguments.

**tl;dr** I'm going to push an update to always have `build` return a promise, but `process.exit` is handled in the webpack plugin.